aioredis/cluster/cluster.py

```diff
@@ -409,76 +395,88 @@ def _execute_node(self, address, command, *args, **kwargs):
         except ReplyError as err:
             address = parse_moved_response_error(err)
             if address is None:
                 raise
             logger.debug('Got MOVED command: {}'.format(err))
             self._moved_count += 1
             if self._moved_count >= self.MAX_MOVED_COUNT:
-                yield from self.initialize()
+                await self.initialize()
                 node = self.get_node(command, *args, **kwargs)
                 address = node.address
-            conn = yield from self.create_connection(address)
+            conn = await self.create_connection(address)
             to_close.append(conn)
-            return (yield from getattr(conn, cmd)(*args, **kwargs))
+            return await getattr(conn, cmd)(*args, **kwargs)
         finally:
             for conn in to_close:
                 conn.close()
-                yield from conn.wait_closed()
+                await conn.wait_closed()
 
-    @asyncio.coroutine
-    def _execute_nodes(self, command, *args, **kwargs):
+    async def _execute_nodes(self, command, *args, all_=False, **kwargs):
         """
         Execute redis command for all nodes and returns
         Future waiting for the answer.
 
         :param command str
+        :param all_ bool - Execute on all nodes masters + slaves
         Raises:
         * TypeError if any of args can not be encoded as bytes.
         * ReplyError on redis '-ERR' responses.
         * ProtocolError when response can not be decoded meaning connection
           is broken.
         """
-        return (yield from asyncio.gather(
-            *[self._execute_node(node.address, command, *args, **kwargs)
-              for node in self._cluster_manager.masters],
-            loop=self._loop))
+        nodes = self.get_nodes_entities(slaves=all_)
+        return await asyncio.gather(*[
+            self._execute_node(node, command, *args, **kwargs)
+            for node in nodes
+        ], loop=self._loop)
 
-    @asyncio.coroutine
-    def execute(self, command, *args, many=False, **kwargs):
+    async def execute(
+            self, command, *args, address=None, many=False, all_=False,
+            **kwargs
+    ):
         """Execute redis command and returns Future waiting for the answer.
 
         :param command str
-        :param many bool - invoke on all nodes
+        :param address tuple - Execute on node with specified address
+            if many specified will be ignored
+        :param many bool - invoke on all master nodes
+        :param all_ boll - if many specified, execute even on slave nodes
         Raises:
         * TypeError if any of args can not be encoded as bytes.
         * ReplyError on redis '-ERR' responses.
         * ProtocolError when response can not be decoded meaning connection
           is broken.
         """
-        if not args:
-            many = True
-        if not many:
+
+        # bad hack to prevent execution on many nodes
```

Comment:
For now, this solution is ok. I plan to use decorators in order to mark commands that
- actually should be executed on all nodes (e.g. flush all)
- should be executed on a random node (e.g. cluster nodes)
- cannot be reasonably executed on a cluster, but must be targeted to a single node (e.g. add slots)

Reply:
Ok, I'll leave it like it is and wait until your changes